internal/resource/nvml-mig-device.go

```diff
@@ -132,3 +133,23 @@ func totalMemory(attr map[string]interface{}) (uint64, error) {
 		return 0, fmt.Errorf("unsupported attribute type %v", t)
 	}
 }
+
+func (d nvmlMigDevice) GetPIEClass() (uint32, error) {
+	info, retVal := d.MigDevice.GetPciInfo()
+	if retVal != nvml.SUCCESS {
+		return 0, retVal
+	}
+	var bytes []byte
+	for _, char := range info.BusId {
+		if char == 0 {
+			break
+		}
+		bytes = append(bytes, byte(char))
+	}
```

Comment:
Instead of implemention this here this should be pulled into `go-nvlib` if it is valid.

Reply:
Happy to do this in a follow-up though.

Reply:
We can decide where to put this behavior once we decide what behavior is expected for MIG devices.

Reply:
Because of the hardcoding of the PCI class for MIG devices this code is removed